src/web/src/app/space/detail/component.ts

```diff
@@ -0,0 +1,280 @@
+	@ViewChild('usersPaginator') usersPaginator: MatPaginator;
+
+	constructor(
+		private route: ActivatedRoute,
+		@Inject(DOCUMENT) private document: any,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private user: UserService,
+	) {
+		this._initInvitationForm();
+		this._initLinkForm();
+	}
+
+	ngOnInit() {
+		this.currentUserId = this.user.info.id;
+
+		const pageSizeStr = localStorage.getItem('spaceDetailPageSize');
+		const pageSize = JSON.parse(pageSizeStr);
+		if (pageSize?.invitations && pageSize?.links && pageSize?.users){
+			this.pageSize = pageSize;
+		}
+
+		this.currentSpaceId = this.route.snapshot.paramMap.get('id');
+		this.getCurrentSpace();
+		this.getInvitations(this.pageSize.invitations, 0);
+		this.getLinks(this.pageSize.links, 0);
+		this.getUsers(this.pageSize.users, 0);
+	}
+
+	getCurrentSpace() {
+		this.space.getById(this.currentSpaceId)
+			.subscribe(res => {
+				this.currentSpace = res;
+			});
+	}
+
+	getInvitations(limit: number, page: number) {
+		this.savePageSize("invitations", limit);
+		this.space.getInvitationListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.invitations = res.results;
+				this.invitationsTotal = res.count;
+			});
+	}
+
+	getLinks(limit: number, page: number) {
+		this.savePageSize("links", limit);
+		this.space.getLinkListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.links = res.results;
+				this.linksTotal = res.count;
+			});
+	}
+
+	getUsers(limit: number, page: number) {
+		this.savePageSize("users", limit);
+		this.space.getUserListForSpace(this.currentSpaceId, limit, page)
+			.subscribe(res => {
+				this.users = res.results;
+				this.usersTotal = res.count;
+			});
+	}
+
+	onLinkCopyBtn(link: SpaceLink){
+		let copyToClipboard = this.document.createElement('textarea');
+		copyToClipboard.style.position = 'fixed';
+		copyToClipboard.style.left = '0';
+		copyToClipboard.style.top = '0';
+		copyToClipboard.style.opacity = '0';
+		copyToClipboard.value = this.document.location.origin + this.linksURL + link.id;
+		document.body.appendChild(copyToClipboard);
+		copyToClipboard.focus();
+		copyToClipboard.select();
+		document.execCommand('copy');
+		document.body.removeChild(copyToClipboard);
+	}
+
+	onInvitationDelBtn(invitation: SpaceInvitation){
+		this.space.delInvitationById(invitation.id)
+			.subscribe(res => {
+				if (this.invitationsPaginator.pageIndex == 0) {
+					this.getInvitations(this.pageSize.invitations, 0);
+				} else {
+					this.invitationsPaginator.firstPage();
+				}			
+			});
+	}
+
+	onLinkDelBtn(link: SpaceLink){
+		this.space.delLinkById(link.id)
+			.subscribe(res => {
+				if (this.linksPaginator.pageIndex == 0) {
+					this.getLinks(this.pageSize.links, 0);
+				} else {
+					this.linksPaginator.firstPage();
+				}			
+			});
+	}
+
+	onUserDelBtn(user: SpaceUser){
+		this.space.delUserById(user.userId)
+			.subscribe(res => {
+				if (this.usersPaginator) {
```

Comment:
Кажется тут должно быть что-то такое:
```suggestion
				if (this.usersPaginator.pageIndex == 0) {
```

Reply:
Поменял. Не досмотрел.